toolchain/lex/tokenized_buffer_test.cpp

```diff
@@ -1107,6 +1107,20 @@ TEST_F(LexerTest, DiagnosticUnrecognizedChar) {
   compile_helper_.GetTokenizedBuffer("\b", &consumer);
 }
 
+TEST_F(LexerTest, DiagnosticFileTooLarge) {
+  Testing::MockDiagnosticConsumer consumer;
+  static constexpr size_t NumLines = 10'000'000;
+  std::string input;
+  input.reserve(NumLines * 3);
+  for (int _ [[maybe_unused]] : llvm::seq(NumLines)) {
+    input += "{}\n";
+  }
+  EXPECT_CALL(consumer, HandleDiagnostic(IsSingleDiagnostic(
+                            DiagnosticKind::TooManyTokens,
+                            DiagnosticLevel::Error, 0x400000, 2, _)));
```

Comment:
It's actually one less than `PayloadBits` because we have two tokens per line. My thinking was that I wanted the test to fail if someone changed `PayloadBits` so that their attention would be drawn to the fact that they're modifying the maximum number of tokens supported (which is otherwise very implicit here). But I think adding a `MaxTokens` would deal with this nicely; I'll do that.

Also I just realized the `2` here is off by one if we want to point at the first "bad" token, because we need to leave room for the end-of-file token.

Reply:
Done.